.github/workflows/.reusable-docker-publish.yml

```diff
@@ -0,0 +1,52 @@
+# reusable workflow
+name: Publish Docker Image
+
+on:
+  workflow_call:
+    inputs:
+      registry-url:
+        type: string
+        description: Github Container Registry base URL
+        required: false
+        default: ghcr.io
+      source-images:
+        type: string
+        description: Source image to publish
+        required: true
+      target-images:
+        type: string
+        description: Target image names
+        required: true
+
+jobs:
+  publish:
+    name: Publish ${{ inputs.source-images }} to ${{ inputs.target-images }}
+    runs-on: ubuntu-latest
+    steps:
+      - name: Login to Github Container Registry
+        uses: docker/login-action@v3
+        with:
+          registry: ${{ inputs.registry-url }}
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Login to Docker Hub
+        uses: docker/login-action@v3
+        with:
+          username: ${{ secrets.DOCKERHUB_USERNAME }}
+          password: ${{ secrets.DOCKERHUB_TOKEN }}
+
+      - name: Docker metadata
+        id: meta
+        uses: docker/metadata-action@v5
+        with:
+          images: ${{ inputs.target-images }}
+          tags: |
+            type=semver,pattern={{version}}
+            type=semver,pattern={{major}}.{{minor}}
+
+      - name: Publish Image
+        uses: contiamo/retag-push@main
+        with:
+          source: ${{ inputs.source-images }}
+          target: ${{ steps.meta.outputs.tags }}
```

Comment:
I'm a little confused as to how this works with multiple source images and multiple targets? Or is the use case only for one of the 2 lists to include more than one element?

Reply:
That's awkward naming from the action author. `target` is, in fact, a list of tags too.